src/Config.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\HttpExtractor;
+
+use Keboola\HttpExtractor\Config\ConfigDefinition;
+use Symfony\Component\Config\Definition\Processor;
+use Symfony\Component\Serializer\Encoder\JsonDecode;
+use Symfony\Component\Serializer\Encoder\JsonEncoder;
+
+class Config
+{
+    /** @var array */
+    private $config;
+
+    private function __construct(
+        array $config
+    ) {
+        $this->config = $config;
+    }
+
+    public static function fromFile(string $configPath): self
```

Comment:
asi nevidim vyhodu oproti obyc ctoru

Reply:
Beru to tak, že ctor by měl popisovat ten nejmenší možný use-case, ideálně jen naplňovat properties a nedělat žádnou logiku okolo. 

Tenhle Config je value object, který dělá jedinou věc - ručí za to, že to co je uvnitř je platný config.  S tím si teď uvědomuju, že je špatně, že načítá jen parameters, asi, ne? Měl by asi mít definici celého toho souboru `config.json` a k tomu mít nějaký ty convenience gettery do parameters (tady jsem trochu opisoval z toho s3extractoru).

Reply:
ja to beru tak, ze nejmensi mozny usecase je ze to nacte config a naplni convenience gettery :) protoze to je jedinej moznej usecase te tridy, ale nehadam se